cmd/memberagent/main.go

```diff
@@ -67,10 +70,32 @@ func init() {
 	metrics.Registry.MustRegister(fleetmetrics.JoinResultMetrics, fleetmetrics.LeaveResultMetrics, fleetmetrics.WorkApplyTime)
 }
 
+var (
+	handleExitFunc = func() {
+		klog.Flush()
+	}
+
+	exitWithErrorFunc = func() {
+		handleExitFunc()
+		os.Exit(1)
+	}
+)
+
 func main() {
 	flag.Parse()
 	utilrand.Seed(time.Now().UnixNano())
-	defer klog.Flush()
+	defer handleExitFunc()
```

Comment:
what's the difference here?

Reply:
right now there's no diff. in case, there are other funcs we won't miss before we exit.